src/Stratis.Features.Collateral/ConsensusRules/VotingRequestFullValidationRule.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Features.PoA;
+using Stratis.Bitcoin.Features.PoA.Voting;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.PoA.Features.Voting;
+using Stratis.Features.Collateral.CounterChain;
+
+namespace Stratis.Features.Collateral.ConsensusRules
+{
+    public class VotingRequestFullValidationRule : FullValidationConsensusRule
+    {
+        private readonly IInitialBlockDownloadState ibdState;
+        private readonly Network network;
+        private readonly Network counterChainNetwork;
+        private readonly JoinFederationRequestEncoder encoder;
+        private readonly VotingManager votingManager;
+        private readonly IFederationManager federationManager;
+
+        public VotingRequestFullValidationRule(IInitialBlockDownloadState ibdState, Network network, CounterChainNetworkWrapper counterChainNetwork, IFederationManager federationManager, VotingManager votingManager) : base()
+        {
+            this.ibdState = ibdState;
+            this.network = network;
+            this.counterChainNetwork = counterChainNetwork.CounterChainNetwork;
+            this.encoder = new JoinFederationRequestEncoder();
+            this.federationManager = federationManager;
+            this.votingManager = votingManager;
+        }
+
+        /// <summary>Checks that any voting requests that are present can be decoded and prohibits re-use of collateral addresses.</summary>
+        public override Task RunAsync(RuleContext context)
+        {
+            if (this.ibdState.IsInitialBlockDownload())
+            {
+                this.Logger.LogTrace("(-)[SKIPPED_IN_IBD]");
+                return Task.CompletedTask;
+            }
+
+            foreach (Transaction transaction in context.ValidationContext.BlockToValidate.Transactions)
+            {
+                CheckTransaction(transaction);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public void CheckTransaction(Transaction transaction)
+        {
+            if (transaction.IsCoinBase || transaction.IsCoinStake)
+                return;
+
+            // This will raise a consensus error if there is a voting request and it can't be decoded.
+            JoinFederationRequest request = JoinFederationRequestBuilder.Deconstruct(transaction, this.encoder);
+            if (request == null)
+                return;
+
+            if (this.federationManager.IsMultisigMember(request.PubKey))
+            {
+                this.Logger.LogTrace("(-)[INVALID_MULTISIG_VOTING]");
+                PoAConsensusErrors.VotingRequestInvalidMultisig.Throw();
```

Comment:
Can you maybe rename this to `MultSigMemberCannotVote` adn the message within that class?

Reply:
Changed to:

`public static ConsensusError InvalidVotingOnMultiSig => new ConsensusError("invalid-voting-on-multisig", "invalid voting on multisig member");`